wasi/README.md

```diff
@@ -0,0 +1,159 @@
+
+### wasip1
+
+Although `wasip1` is marked as "legacy", this is the version that nearly all compilers support when compiling to a Wasm target. For that reason we are aiming to have good support, but have no immediate plans to "complete" the implementation.
+Please reach out if you'd like to see more done to support wasip1.
+
+Here are some features we have basic coverage for currently:
+
+* stdin / stdout / stderr
+* environment variables
+* command arguments
+* clocks
+* random
+* basic reading and writing of files (through use of a [virtual file system](https://github.com/google/jimfs))
+
+If your module calls a wasi function that we don't support, or uses a feature that we don't support, we will throw a `WASMRuntimeException`.
+
+For the most up-to-date info, and to see what specific functions we support, see the [WasiPreview1.java](https://github.com/dylibso/chicory/blob/main/wasi/src/main/java/com/dylibso/chicory/wasi/WasiPreview1.java) class.
+We also have a table:
+
+| WASI Function            | Supported |
+|--------------------------|-----------|
+| args_get                 | ✅         |
+| args_sizes_get           | ✅         |
+| clock_res_get	           | ❌         |
+| clock_time_get           | 👷        |
+| environ_get	             | ✅	        |
+| environ_sizes_get        | 	✅	       |
+| fd_advise		              | ❌         |
+| fd_allocate	             | ❌         |
+| fd_close	                | ✅         |
+| fd_datasync	             | ❌         |
+| fd_fdstat_get            | 	✅        |
+| fd_fdstat_set_flags      | 	✅        |
+| fd_fdstat_set_rights	    | 💀         |
+| fd_filestat_get	         | ✅	        |
+| fd_filestat_set_size     | 	❌	       |
+| fd_filestat_set_times	   | ❌         |
+| fd_pread	                | ❌         |
+| fd_prestat_dir_name      | 	✅	       |
+| fd_prestat_get           | 	✅	       |
+| fd_pwrite	               | ❌	        |
+| fd_read	                 | ✅	        |
+| fd_readdir	              | ✅         |
+| fd_renumber	             | ❌         |
+| fd_seek	                 | ✅	        |
+| fd_sync	                 | ❌         |
+| fd_tell	                 | ✅	        |
+| fd_write	                | ✅         |
+| get_allocation_state	    | 💀        |
+| get_state_ptr	           | 💀        |
+| memcpy	                  | 💀        |
+| memmove	                 | 💀        |
+| memset	                  | 💀        |
+| path_create_directory	   | ✅         |
+| path_filestat_get        | 	✅	       |
+| path_filestat_set_times	 | ❌         |
+| path_link	               | ❌         |
+| path_open	               | 👷        |
+| path_readlink	           | 👷        |
+| path_remove_directory    | 	✅	       |
+| path_rename              | 	✅	       |
+| path_symlink	            | ❌         |
+| path_unlink_file         | 	✅	       |
+| poll_oneoff	             | ❌         |
+| proc_exit                | 	✅	       |
+| proc_raise	              | 💀        |
+| random_get               | 	✅	       |
```

Comment:
We might note that the maximum size is limited, to `100_000` bytes.

(arbitrary limit to limit resource usage ... we could change this to use a loop that checks for interruption)